### ISSUE TITLE
Receiving phishing emails to my attached contact email

### ISSUE THREAD

**Issue by 17sdheeraj** [is-a-dev/register#14802]
Hello!
I have seen that since a few days I have been receiving phishing emails to the email I added to my file (contact-sdheeraj-isadev@domain.com). I have made this to ask if anyone else has been receiving these type of emails.
Some screenshots of the emails:
![image](https://github.com/is-a-dev/register/assets/131849765/ae914737-2f7f-4f38-9e37-56d16b6f9166)
![image](https://github.com/is-a-dev/register/assets/131849765/e2ee3ac9-9be6-4afd-a3cb-b60aa7c6f7c6)
And more were there which got rejected and were not delivered to me
### My subdomain
https://sdheeraj.is-a.dev

**Comment by CuteDog5695** [comment]
Wait... I've also gotten emails from the umm.de domain. Is it possible that somebody is emailing everyone who has a domain here?

**Comment by Stef-00012** [comment]
same here

![Screenshot_20240710_191711_Gmail](https://github.com/is-a-dev/register/assets/81536172/a6e60b03-0c98-4dcd-ac07-83fd56457c7c)

**Comment by xzonix** [comment]
I got the exact same mail from a charity of Germany telling me that i received a private donation.
![image](https://github.com/is-a-dev/register/assets/112401813/41c5397b-a10e-4441-abe4-18d6eb0725ca)

**Comment by xzonix** [comment]
That email got into my second gmail account, and i didn't introduce this mail to is-a.dev
![image](https://github.com/is-a-dev/register/assets/112401813/98a3fe0b-395e-40ca-bc4f-f37744c40c93)

**Comment by wdhdev** [comment]
Would it be worth dropping emails in the owner key entirely and just relying on the commit history?

**Comment by xzonix** [comment]
> The spams are getting more and more day by day
> ![image](https://github.com/user-attachments/assets/2986c401-3d0e-4184-8962-955b03669620)
> 

Yo bro, this also happened to me, luckily, i acted fast and removed my email from the JSON file and didn't got more emails. Just leave "mail" in blank and add Discord to the JSON.

**Comment by wdhdev** [comment]
@phenax Could we use an external DB of some sorts?

**Comment by phenax** [comment]
Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.

But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.

We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.

@is-a-dev/maintainers, what do you all think?

**Comment by xzonix** [comment]
> > > The spams are getting more and more day by day
> > > ![image](https://private-user-images.githubusercontent.com/131849765/348054560-2986c401-3d0e-4184-8962-955b03669620.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3ODk2ODksIm5iZiI6MTcyMDc4OTM4OSwicGF0aCI6Ii8xMzE4NDk3NjUvMzQ4MDU0NTYwLTI5ODZjNDAxLTNkMGUtNDE4NC04OTYyLTk1NWIwMzY2OTYyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxMlQxMzAzMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Mjc1NzIzN2ZiODRlOTUzNTU3ZjM4MjFhZmNlMWNhNmQzYzcwZTZiNTVjMjkwMTFkNDZiNjllY2JjMjg1Njc2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bwDui8PnDndq4e9HOITYF7gLnNSVFCFcQH9ZJx75O7E)
> > 
> > Yo bro, this also happened to me, luckily, i acted fast and removed my email from the JSON file and didn't got more emails. Just leave "mail" in blank and add Discord to the JSON.
> 
> If that is the case that means they are fetching emails from github instead of scraping and storing them

Probably they are.

**Comment by DEV-DIBSTER** [comment]
> Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.
> 
> But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.
> 
> We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.
> 
> @is-a-dev/maintainers, what do you all think?

Damage has already been done sadly but yes, I do think we should:

1) Remove the email field.
2) Switch over to Discord ID contact.
3) Hope for the best from the changes.

**Comment by xzonix** [comment]
> > Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.
> > 
> > But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.
> > 
> > We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.
> > 
> > @is-a-dev/maintainers, what do you all think?
> 
> Damage has already been done sadly but yes, I do think we should:
> 
> 1) Remove the email field.
> 2) Switch over to Discord ID contact.
> 3) Hope for the best from the changes.

I definitely agree

**Comment by wdhdev** [comment]
> Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.

I created https://data.is-a.dev a few months to a year ago basically to prove how is-a.dev is literally just a data farm for scammers.

> But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.

Yeah that would work, however what would we do with existing domains, just only have GitHub usernames? Also this brings up another issue, what do we do with domains where the original author's account has been deleted, and what do we do with username changes, because we can't exactly rely on people to immediately update their info.

> We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.

Yeah not much we can do about that, however removing them all from the main repo would help.

**Comment by creeperita09** [comment]
Yep me got the email in the second screenshot

**Comment by xzonix** [comment]
> @0v90 's suggestion which fell into my DMs
> ![image](https://github.com/user-attachments/assets/3dca8550-0f15-45f8-b2a4-8ded4555275c)
> 

Another idea would be, that you must provide the email on the is-a.dev website, and when you send the email/data, it automatically sends it to data.is-a.dev instead of doing it via github jsons.

**Comment by xzonix** [comment]
> > > @0v90 's suggestion which fell into my DMs
> > > ![image](https://private-user-images.githubusercontent.com/70831061/348283800-3dca8550-0f15-45f8-b2a4-8ded4555275c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3OTI3MjEsIm5iZiI6MTcyMDc5MjQyMSwicGF0aCI6Ii83MDgzMTA2MS8zNDgyODM4MDAtM2RjYTg1NTAtMGYxNS00NWY4LWIyYTQtOGRlZDQ1NTUyNzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEyVDEzNTM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlZjcxY2NmNmY2YTM1Mzc3MmNhY2ZjMDc1ZjliZjVmYTQxY2UxMzE1NjBkMzE3NjMzZDYwZGU1Zjk5NzYyNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EfuV6wy05JGprNQQXnvxMk35p5JQl9WAEFGxl592N2k)
> > 
> > Another idea would be, that you must provide the email on the is-a.dev website, and when you send the email/data, it automatically sends it to data.is-a.dev instead of doing it via github jsons.
> 
> Yea this idea is good but it would take a little time to code it and stuff

Yes, it can be hard to code, but it's for yall's security

**Comment by phenax** [comment]
> Another idea would be, that you must provide the email on the is-a.dev website, and when you send the email/data, it automatically sends it to data.is-a.dev instead of doing it via github jsons.

I think this introduces a weird bit of complexity that's better avoided. Although if we all agree that having users' email address is worth that complexity then we can go with this.


> however what would we do with existing domains, just only have GitHub usernames?

We can send people an email letting them know that we're removing email addresses and to update their contacts. If they do that or don't is up to them.


But I would still like to point out that once we remove all the email addresses, whats stopping someone from going 1 commit before the change. Even if we rebase everything since the dawn of time, any recent forks can still be used. We should still do it but if we can think of a solution to obscure it better, we should consider it.

On a side note, what if we screw with people scraping this information a bit? Change the emails to point to nothing, making them unusable but still giving them a false sense that this is reaching someone.

**Comment by 0v90** [comment]
> > Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.
> > But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.
> > We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.
> > @is-a-dev/maintainers, what do you all think?
> 
> Damage has already been done sadly but yes, I do think we should:
> 
> 1. Remove the email field.
> 2. Switch over to Discord ID contact.
> 3. Hope for the best from the changes.

I generally think any direct showen revealed "social contact" that directy links to any person is any good for their privacy , rather to be encrypted and maybe even given a unique ID for the user themselves , so none of their info is exposed publicly

**Comment by 0v90** [comment]
> > Another idea would be, that you must provide the email on the is-a.dev website, and when you send the email/data, it automatically sends it to data.is-a.dev instead of doing it via github jsons.
> 
> I think this introduces a weird bit of complexity that's better avoided. Although if we all agree that having users' email address is worth that complexity then we can go with this.
> 
> > however what would we do with existing domains, just only have GitHub usernames?
> 
> We can send people an email letting them know that we're removing email addresses and to update their contacts. If they do that or don't is up to them.
> 
> But I would still like to point out that once we remove all the email addresses, whats stopping someone from going 1 commit before the change. Even if we rebase everything since the dawn of time, any recent forks can still be used. We should still do it but if we can think of a solution to obscure it better, we should consider it.
> 
> On a side note, what if we screw with people scraping this information a bit? Change the emails to point to nothing, making them unusable but still giving them a false sense that this is reaching someone.

For now the most important thing if focusing on the secuirty/pivacy part of the current users , and messing up with those who is ever behind the phising attacks may come later after dealing with point 1 most importantly!

**Comment by 0v90** [comment]
> suggestion

It was never said to be easy , it will indeed be complex , getting the data encrypting them , giving them a unique ID and so on , that actually can be the easy part , the hard one is storing them in data bases and actually working with the data base its self , which for sure will be a big of a hustle specially with hundered of thousands of users or ever more! , but is it worth it , that's really up to you to evalute it the way you want , but IMO it may be one of the ways to secure the user info , and make no info regards to them publicly exposed to anyone , rather just the devs of the project. 

This is just one idea , there is definitely more ideas and even more better ones.

**Comment by axviii3** [comment]
I dont know much about github actions... but maybe a github action which parses the json while merging and replaces the contact details with a public encryption key and the service or who ever needs to access the details can decrypt it with a private key? I dont know how much work this would take though.

Other options are using alternative methods to store the info, contacting mergers via github credentials (big maybe), some sort of sign in via discord, github or other auth and finally (for me) using some form of registration platform like a discord bot (or a website like there previously was) that can carefully store that data

**Comment by DEV-DIBSTER** [comment]
> > > Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.
> > > But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.
> > > We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.
> > > @is-a-dev/maintainers, what do you all think?
> > 
> > 
> > Damage has already been done sadly but yes, I do think we should:
> > 
> > 1. Remove the email field.
> > 2. Switch over to Discord ID contact.
> > 3. Hope for the best from the changes.
> 
> I generally think any direct showen revealed "social contact" that directy links to any person is any good for their privacy , rather to be encrypted and maybe even given a unique ID for the user themselves , so none of their info is exposed publicly

Why I understand where you're getting at with this (less user information publicly) I believe as a temporary and easy solution is to lock it down to a platform which can limit any form of direct spam. 

An email is an email, anyone who get's their hands on someone's email can send mail to it, or sign it up for newsletters, and nothing is stopping these sites or scammers from sending it. Nobody sends newsletters through Discord. The worst thing in my opinion is either a tiny increase in friend requests or message requests. That's it. 

We can think of long term solutions here, but as a quick fix, just limit it to a Discord ID. Every staff member here has Discord from what I'm aware of. Yeah that's my two cents.

**Comment by 0v90** [comment]
> > > > Let's not ask for email anymore? It was a bad idea to ask for that in the first place. Who came up this terrible system? Oh right, it was me.
> > > > But I think any reliable means to contact them is more than good enough and since a lot of our users are also on discord, that seems like a good default. If not that, twitter, mastodon, etc. works just as well. At least 1 more way to get in touch other than github. If all else fails, we still have the github username to tag them on issues or discussions.
> > > > We can also remove all existing emails or encrypt them in-place but that information is already spread accross thousands of forks and all PRs so not sure how much that helps.
> > > > @is-a-dev/maintainers, what do you all think?
> > > 
> > > 
> > > Damage has already been done sadly but yes, I do think we should:
> > > 
> > > 1. Remove the email field.
> > > 2. Switch over to Discord ID contact.
> > > 3. Hope for the best from the changes.
> > 
> > 
> > I generally think any direct showen revealed "social contact" that directy links to any person is any good for their privacy , rather to be encrypted and maybe even given a unique ID for the user themselves , so none of their info is exposed publicly
> 
> Why I understand where you're getting at with this (less user information publicly) I believe as a temporary and easy solution is to lock it down to a platform which can limit any form of direct spam.
> 
> An email is an email, anyone who get's their hands on someone's email can send mail to it, or sign it up for newsletters, and nothing is stopping these sites or scammers from sending it. Nobody sends newsletters through Discord. The worst thing in my opinion is either a tiny increase in friend requests or message requests. That's it.
> 
> We can think of long term solutions here, but as a quick fix, just limit it to a Discord ID. Every staff member here has Discord from what I'm aware of. Yeah that's my two cents.


I can still write a " friend request " spam bots to all these ID's, but yes the spam messages would be none to few if the person accept few of these friend requests and message dms, but still , another idea but not completely secure/private for the user info. The idea is indeed limiting and way less limiting than emails , but how limiting it's overall and actually how affective will it be that my self I don't know either, I guess we would have to wait and see if the idea to be considered.

**Comment by wdhdev** [comment]
> But I would still like to point out that once we remove all the email addresses, whats stopping someone from going 1 commit before the change. Even if we rebase everything since the dawn of time, any recent forks can still be used. We should still do it but if we can think of a solution to obscure it better, we should consider it.

Maybe @github-staff could somehow purge all forks and rebase the entire repo? It would be a bit complicated but it would do most of the work for us.

**Comment by wdhdev** [comment]
Personally I think we should setup some unique ID based system (is there some sort of OSS that does this?) as multiple users in this thread have suggested.

The owner key could be updated from an object to just a string value like this:
```json
{
  "owner": "k2H9rSQ6KB2373b3FeUR28WX8RxaZvn6"
}
```

We could probably (and fairly easily...) create a script to register all existing owner information with a database which then returns a custom ID like the one above and it will just update all domains to be like this.

**Comment by wdhdev** [comment]
> For that I think they can make a discord bot which generates a id for the people so they can use it in their json

That could be good. Like it encrypts their details with a hash/salt that can be decrypted using a master key?

**Comment by wdhdev** [comment]
Yeah that was my idea.

**Comment by wdhdev** [comment]
![Screenshot_20240716_165016_Outlook](https://github.com/user-attachments/assets/4aec3b92-8a61-4a10-85e9-47642b228566)

**Comment by andrewstech** [comment]
https://github.com/is-a-dev/owl

**Comment by wdhdev** [comment]
> This looks cool but can we have an option to edit the email address as many people do not like giving their main email address and instead give their alt email address.

If you have other email addresses on your GitHub account they will show up there.

**Comment by wdhdev** [comment]
@phenax @andrewstech We need a solution for this ASAP. I'm personally receiving multiple phishing/scam emails per day.

**Comment by orxngc** [comment]
> I'm personally receiving multiple phishing/scam emails per day.

Same

**Comment by Arch881010** [comment]
I think the owner object should get phased out for the ID as a string.
Or maybe 
```json
"owner": {
  "ID":"<owl ID">
  }
  ```
  and have the old fields still applicable.

**Comment by wdhdev** [comment]
Would be better as a string instead of a key, however for backwards compatibility it might be better as a key and just added as a new field.

**Comment by wdhdev** [comment]
> I have removed all my domains which have my email on it on Jun 11 and never received a spam email~

They are most likely fetching the most recent commit on the repo.

> I investigated the domains that this emails come from. most likely those are universities stolen emails or students from those institutes?

I would believe it's weak email security from the university with a combination of weak passwords from the students causing emails to get hacked.

**Comment by wdhdev** [comment]
I believe I've found the cause of the issue, I think the scammers have been using the [Raw API](https://github.com/is-a-dev/raw-api) to fetch the emails. I have redacted all emails from the Raw API.

Let's see if this makes any difference in the amount of scam emails. If it seems to be solved I'll most likely close this issue.

**Comment by andrewstech** [comment]
> the

The owl project is deployed Im just waiting on you again :(

**Comment by andrewstech** [comment]
OWL is now fully deployed and integrated into the discord bot or is available at https://owl.is-a.dev/. There is 24 records currently using owl docs will be posted soon
